src/etc/inc/captiveportal.inc

```diff
@@ -1584,46 +1493,158 @@ function captiveportal_syslog($message) {
-		$radiusctx = 'first';
+	/* Unable to find the any authentication server config - shouldn't happen! - bail out */
+	if(count($auth_servers) === 0) {
+		log_error("Zone: {$cpzone} - No valid server could be used for authentication.");
+		$reply_message = gettext("Internal Error");
 	}
-
-	$auth_list = RADIUS_AUTHENTICATION($username,
-		$password,
-		$radiusservers[$radiusctx],
-		$clientip,
-		$clientmac,
-		$pipeno);
-
-	if ($auth_list['auth_val'] == 2) {
-		captiveportal_logportalauth($username, $clientmac, $clientip, $type);
-		$sessionid = portal_allow($clientip,
-			$clientmac,
-			$username,
-			$password,
-			$auth_list,
-			$pipeno,
-			$radiusctx);
-	} else {
-		captiveportal_free_dn_ruleno($pipeno);
+	else {
+		foreach($auth_servers as $authcfg) {
+			if($authlevel < 3) {
+				$radmac_error = false;
+				$attributes = array("nas_identifier" => "CaptivePortal",
+					"nas_port_type" => RADIUS_ETHERNET,
+					"nas_port" => $pipeno,
+					"framed_ip" => $clientip);
+				if(mac_format($clientmac) !== null) {
+					$attributes["calling_station_id"] = mac_format($clientmac);
+				}
+				
+				$result = null;
+				$status = null;
+				$msg = null;
+				
+				/* Radius MAC authentication */
+				if($cpcfg['auth_method'] === 'radmac' && $clientmac) {
+					if($authcfg['type'] === 'radius') {
+						$login = mac_format($clientmac);
+						$password = $cpcfg['radmac_secret'];
+						$status = "MACHINE LOGIN";
+					} else {
+						/* Trying to perform a Radius MAC authentication on a non-radius server - shouldn't happen! - bail out */
+						$msg = gettext("Internal Error");
+						log_error("Zone: {$cpzone} - Trying to perform RADIUS MAC authentication on a non-RADIUS server !");
+						$radmac_error = true;
+						$result = null;
+					}
+				}
+				
+				if(!$radmac_error) {
+					if($authcfg['type'] === 'none') {
+						$result = true;
+					} else {
+						$result = authenticate_user($login, $password, $authcfg, $attributes);
+					}
+					
+					if(!empty($attributes['error_message'])) {
+						$msg = $attributes['error_message'];
+					}
+					
+					if ($authcfg['type'] == 'Local Auth' && $result && isset($cpcfg['localauth_priv'])) {
+						if(!userHasPrivilege(getUserEntry($login), "user-services-captiveportal-login")) {
+							$result = false;
+							$msg = gettext("Access Denied");
+						}
+					}
+					
+					if(empty($status)) {	
+						if($result === true) {
+							$status = "ACCEPT";
+						} else if($result === null) {
+							$status = "ERROR";
+						} else {
+							$status = "FAILURE";
+						}
+					}
+					
+					if($cpcfg['auth_method'] === 'radmac' && $login == mac_format($clientmac) || $authcfg['type'] === 'none' && empty($login)) {
+						$login = "unauthenticated";
+					} 
+					// We determine a flag
+					if($result === true) {
+						$val = 3;
+					}
+					else if($result === false && !empty($attributes['reply_message'])) {
+						$val = 2;
+						$msg = $attributes['reply_message'];
+					}
```

Comment:
Change to `elseif`, move up to the same line as the closing brace, add space after `elseif`.